### ISSUE TITLE
MySQL FIELD function

### ISSUE THREAD

**Issue by dvandersluis** [Faveod/arel-extensions#12]
I created an extension for handling MySQL's `FIELD` function, would that be valuable as a PR? As far as I can tell, there's no parallel functions in the other RDBMSes (although it could maybe be faked with a CASE statement?), so I'm not sure if there's a way to handle non-universally supported functions, or a desire for them in this gem.

```ruby
ids = [1, 2, 3, 4, 5]
Post.order(Post[:id].field(ids)).to_sql
=> "SELECT `posts`.* FROM `posts` ORDER BY FIELD(`posts`.`id`, 1, 2, 3, 4, 5)"
```

**Comment by jdelporte** [comment]
Hi Daniel,
If you can propose us an alternative for the different DBMSes, we will gladly accept such PR. 
In the meantime, if you are sure to be using MySQL, you can use the node NamedFunction :
``` ruby
Arel::Nodes::NamedFunction.new('FIELD',[Post.arel_table[:id],1,2,3,4,5]).to_sql
=> "FIELD(`posts`.`id`, 1, 2, 3, 4, 5)"
```

**Comment by dvandersluis** [comment]
Yeah, I know about `NamedFunction` but I like to implement things we use frequently as extensions to `arel-extensions` because the syntax is much nicer IMO :smile:

I'll try to figure out how I can implement `FIELD` for the other DBMSes.
